### ISSUE TITLE
🪟 🐛  "Discard changes" modal appears in the wrong place during connection creation

### ISSUE THREAD

**Issue by dizel852** [airbytehq/airbyte#15515]
## Environment
- **Airbyte version**:  0.40.0-alpha
- **OS Version / Instance**:  macOS
- **Deployment**: local deployment
- **Step where error happened**: Connections / New connection  / Select source 
## Current Behavior
"Discard changes" modal appears if a user has chosen one of the sources in "Set up the source" form, but didn't actually fill out none of the fields, and then decided to choose the existing source.

![CPT2208101840-1229x809](https://user-images.githubusercontent.com/20929344/183948165-d92738fc-4c76-4b10-a771-89d2704c00ab.gif)

## Expected Behavior
I am pretty sure the user would not expect to see the "Discard changes" modal, even if he changed his mind and decided to choose the "Source" from the existing ones.

## Steps to Reproduce
**Prerequisites**: you need to have at least one existing source

1. Open the "Connections" page from the left side panel
2. Click "New Connection" button
3. On "Set up the source" form select any provided source from the dropdown.
4. On "Select an existing source" form select one of the existing sources from the dropdown.
5. Click "Use existing source button"

## Are you willing to submit a PR?
yes

**Comment by octavia-squidington-iii** [comment]
cc @airbytehq/frontend

**Comment by tealjulia** [comment]
I believe #14583 is a related bug.

**Comment by dizel852** [comment]
@tealjulia oh yeah, thanks! Will close it as a duplicate of [#14583](https://github.com/airbytehq/airbyte/issues/14583) 👍🏻
